### ISSUE TITLE
Merging linear paths : computed sequence length and computed LN differ

### ISSUE THREAD

**Issue by AntoineHo** [ggonnella/gfapy#20]
Hello,

I am trying to merge linear paths in a GFA after removing some segments (and their associated links).

However, I got this error : `'str' object has no attribute 'length'`

After some looking into it, I think the error is not the actual error. It is due to this condition in line 335 of the linear_paths.py file `merged.LN != len(merged.sequence):`

Since `merged.LN` and `len(merged.sequence)` were different, it should raise the `InconsistencyError` but it raises the aforementioned "no length attribute error" instead. I think it is because in the lines 336-338, the format call for `merged.sequence.length` instead of `len(merged.sequence)` :

```python
"""Currently : raises Error 'str' object has no attribute 'length'"""

        elif self._vlevel > 0 and merged.LN != len(merged.sequence):
          raise gfapy.InconsistencyError(
              "Computed sequence length {} ".format(merged.sequence.length)+
              "and computed LN {} differ".format(merged.LN))

"""Correction, that now raises : Error Computed sequence length 15968 and computed LN 15386 differ"""

        elif self._vlevel > 0 and merged.LN != len(merged.sequence):
          raise gfapy.InconsistencyError(
              "Computed sequence length {} ".format(len(merged.sequence))+
              "and computed LN {} differ".format(merged.LN))
```

Now, when I try, I get the error : `Computed sequence length 15968 and computed LN 15386 differ`. Do you know why the merging creates this difference (here 582 bp) and how to avoid it (besides putting the vlevel to 0) ?

I do not know if this is relevant but, I get this error only with some GFA produced by the Raven assembler whereas with the same input reads but using other assemblers (such as miniasm or Flye), the merging works fine.

EDIT:

After looking at this issue further, I found that it happens in cases such as the example below. Two segments are linked but the CIGAR ni the link has a longer match than the length of one of the two linked sequences. Here : 7M > length of SegB

```plain
S       SegA    AAAAAAAAAA      LN:i:10 RC:i:1
S       SegB    TTTTT   LN:i:5  RC:i:1
L       SegA    +       SegB    -       7M
```

The following code:
```python
import gfapy
import sys
gfa = gfapy.Gfa.from_file(sys.argv[1])
newgfa = gfa.merge_linear_paths()
[0] gfapy.error.InconsistencyError: Computed sequence length 10 and computed LN 8 differ
```
Interestingly, when using `gfapy-mergelinear`, the error is not raised and the merged GFA contains one segment with LN:i:8 when the reported sequence is actually 10 base long. I think this is because the vlevel is set to 0 when using `gfapy-mergelinear` whereas is was set to default (I guess vlevel=1) in the above python script.

```
S       1       AAAAAAAAAA      LN:i:8
```
I do not know which behavior is expected in such cases.

Cheers

**Comment by ggonnella** [comment]
Hi @AntoineHo
Thank you very much for reporting this. I fixed the bug in the error messages. Well spotted!

Regarding the example: I have troubles to understand what the Link with 7M means, if one of the sequences is just 5 bp. The L is supposed to represent dovetail overlaps. That makes the line incorrect... Therefore I am not sure, what gfapy should do in this case... Your suggestion?

**Comment by AntoineHo** [comment]
Hi,

Actually I found out that this is due to how Raven computes the overlaps. I asked the authors and they replied that because raven uses of minimizers to compute the overlap, when some large indel are present between the segments, it creates these erroneous lines.

To avoid raising the Inconsistency error, I simply checked all the links in the GFA and disconnected them when I found an overlap longer than one of the segments. Ideally I think it could be possible to realign the segments and produce a good overlap but I am not sure how to do this and this seems like too much work for these rare cases.

Thanks for your reply.
Cheers
